### PR TITLE
Rename `DialogTitle` to `DialogHeader`

### PR DESCRIPTION
**Pull request by lukeac123**
Rename DialogTitle to DialogHeader
Rename title and subtitle props to header and preheader
Add patch for mosaic content editor

### CODE REVIEW

File: packages/lab/src/dialog/DialogHeader.tsx [hunk 1]
```diff
@@ -68,12 +68,10 @@ export const DialogTitle = ({
     >
       {status && <StatusIndicator status={status} />}
       <div>
-        {subtitle && (
-          <Text as={"label"} variant="secondary">
-            {subtitle}
-          </Text>
+        {preheader && (
+          <h3 className={withBaseName("preheader")}>{preheader}</h3>
```
Comment (lukeac123) [inline packages/lab/src/dialog/DialogHeader.tsx:72]: semantically think this makes sense as a h3 ?

File: package.json
Comment (lukeac123) [inline package.json:1]: patch needed to update the renaming in the mosaic-content-editor package

File: packages/lab/src/dialog/DialogHeader.tsx (hunk 1 above)
Comment (joshwooding) [reply packages/lab/src/dialog/DialogHeader.tsx:72]: A H3 isn't appropriate here, this would make a h3 appear before a h2

Comment (lukeac123) [reply packages/lab/src/dialog/DialogHeader.tsx]: changed to a div

File: site/docs/components/dialog/examples.mdx
[the remark this answers is not in the document]
Comment (lukeac123) [reply site/docs/components/dialog/examples.mdx:1]: thanks - swapped all references to title to header

**Review by lilyvc — APPROVED** [review]

**Pull request merged**